### PR TITLE
Ad Hoc Charge & (Category) api docs

### PR DESCRIPTION
**Pull request by HenryTraill**

### CODE REVIEW

File: pages/adhoccharges/ad-hoc-charge-object.json [hunk 1]
```diff
@@ -2,6 +2,18 @@
...
+    "branch_tax_setup": null,
+    "charge_via_branch": false,
+    "contractor_tax_setup": null,
+    "contractor_usable": false,
+    "default_charge_amount": null,
+    "default_description": null,
+    "default_pay_amount": null,
+    "dft_net_gross": "gross",
+    "id": 11,
+    "name": "Registration fee"
```
Comment (alextford11) [inline pages/adhoccharges/ad-hoc-charge-object.json:15]: Try to use example as much as possible please, strings and integers are easy to do, FK i don't care

File: pages/adhoccharges/adhoccharges.yml [hunk 2]
```diff
@@ -22,3 +22,29 @@ sections:
...
+  -
+    title: Delete an Ad Hoc Charge
+    id: delete-an-ad-hoc-charge
+    description: /adhoccharges/delete-an-ad-hoc-charge.md
+    code: /adhoccharges/delete-an-ad-hoc-charge.py
+    code_type: DELETE
+    code_url: /api/adhoccharges/<id>/
+    response: /adhoccharges/delete-an-ad-hoc-charge.json
+
+
```
Comment (alextford11) [inline pages/adhoccharges/adhoccharges.yml:50]: remove extra line

File: pages/adhoccharges/create-an-ad-hoc-charge.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    "branch_tax_setup": null,
+    "charge_via_branch": false,
+    "contractor_tax_setup": null,
+    "contractor_usable": false,
+    "default_charge_amount": null,
+    "default_description": null,
+    "default_pay_amount": null,
+    "dft_net_gross": "gross",
+    "id": 7,
+    "name": "Registration fee"
```
Comment (alextford11) [inline pages/adhoccharges/create-an-ad-hoc-charge.json:15]: more examples plz

Comment (alextford11) [reply pages/adhoccharges/create-an-ad-hoc-charge.json:15]: just use the same all over and copy and paste it in

File: pages/adhoccharges/create-an-ad-hoc-charge.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+### Create an Ad Hoc Charge
+
+Creating an Ad Hoc Charge can be completed by simply supplying the `date_occurred`, `category` and either both 
+`charge_client` and `client` or `pay_contractor` and `contractor`.
```
Comment (alextford11) [inline pages/adhoccharges/create-an-ad-hoc-charge.md:4]: description?

File: pages/adhoccharges/create-an-ad-hoc-charge.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import pprint, requests, datetime
+
+headers = {'Authorization': 'token <API KEY>'}
+data = {
+    'service': 26,
+    'client': 37,
+    'charge_client': 25,
+    'category': 7,
+    'date_occurred': datetime.datetime(2021, 1, 1),
```
Comment (alextford11) [inline pages/adhoccharges/create-an-ad-hoc-charge.py:9]: description?

File: pages/adhoccharges/get-an-ad-hoc-charge.json [hunk 6]
```diff
@@ -2,6 +2,18 @@
   "id": 32,
   "agent": null,
   "appointment": null,
+  "category": {
```
Comment (alextford11) [inline pages/adhoccharges/get-an-ad-hoc-charge.json:5]: !

File: pages/adhoccharges/update-an-ad-hoc-charge.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
+{
+  "id": 32,
+  "agent": null,
+  "appointment": null,
+  "category": {
```
Comment (alextford11) [inline pages/adhoccharges/update-an-ad-hoc-charge.json:5]: !

File: pages/adhoccharges/update-an-ad-hoc-charge.md [hunk 8]
```diff
@@ -0,0 +1,4 @@
+### Update an Ad Hoc Charge
+
+Updating an Ad Hoc Charge can be completed by simply supplying the `date_occurred`, `category`, any fields to
```
Comment (alextford11) [inline pages/adhoccharges/update-an-ad-hoc-charge.md:3]: description?

File: pages/adhoccharges/update-an-ad-hoc-charge.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import pprint, requests, datetime
+
+headers = {'Authorization': 'token <API KEY>'}
+data = {
+    'service': 26,
```
Comment (alextford11) [inline pages/adhoccharges/update-an-ad-hoc-charge.py:5]: description?

File: pages/ahc-categories/ad-hoc-charge-category-object.json [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "Registration fee",
+  "branch_tax_setup": null,
+  "charge_via_branch": false,
+  "contractor_tax_setup": null,
+  "contractor_usable": false,
+  "default_charge_amount": null,
+  "default_description": null,
+  "default_pay_amount": null,
+  "dft_net_gross": "gross"
+}
```
Comment (alextford11) [inline pages/ahc-categories/ad-hoc-charge-category-object.json:12]: new line

File: pages/ahc-categories/ad-hoc-charge-category-object.json [hunk 11]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (alextford11) [inline pages/ahc-categories/ad-hoc-charge-category-object.json:1]: !

File: pages/ahc-categories/ad-hoc-charge-category-object.md [hunk 12]
```diff
@@ -0,0 +1,4 @@
+### Get an Ad Hoc Charge Category
+
+Returns the details of an existing ah hoc charge category. You only need to specify the unique
```
Comment (alextford11) [inline pages/ahc-categories/ad-hoc-charge-category-object.md:3]:
```suggestion
Returns the details of an existing Ad Hoc Charge Category. You only need to specify the unique
```

File: pages/ahc-categories/ad-hoc-charge-category-object.md [hunk 13]
```diff
@@ -0,0 +1,4 @@
...
+`id` of the ad hoc charge category to get the correct details.
```
Comment (alextford11) [inline pages/ahc-categories/ad-hoc-charge-category-object.md:4]:
```suggestion
`id` of the Ad Hoc Charge Category to get the correct details.
```

File: pages/ahc-categories/list-all-ad-hoc-charge-categories.json [hunk 14]
```diff
@@ -0,0 +1,20 @@
+{
+  "count": 32,
+  "next": null,
+  "previous": null,
+  "results": [
+    {
```
Comment (alextford11) [inline pages/ahc-categories/list-all-ad-hoc-charge-categories.json:6]: !

File: pages/ahc-categories/list-all-ad-hoc-charge-categories.md [hunk 15]
```diff
@@ -0,0 +1,4 @@
+### List all Ad Hoc Charges Categories
+
+Returns a list of your ad hoc charge categories. The ad hoc charge categories are sorted by `id`, with
```
Comment (alextford11) [inline pages/ahc-categories/list-all-ad-hoc-charge-categories.md:3]:
```suggestion
Returns a list of your Ad Hoc Charge Categories. The Ad Hoc Charge Categories are sorted by `id`, with 
```

File: pages/ahc-categories/list-all-ad-hoc-charge-categories.md [hunk 16]
```diff
@@ -0,0 +1,4 @@
+### List all Ad Hoc Charges Categories
+
+Returns a list of your ad hoc charge categories. The ad hoc charge categories are sorted by `id`, with 
+the largest id first.
```
Comment (alextford11) [inline pages/ahc-categories/list-all-ad-hoc-charge-categories.md:4]: is it?

**Review by alextford11 — CHANGES_REQUESTED** [review]

File: pages/ahc-categories/list-all-ad-hoc-charge-categories.md (hunk 16 above)
Comment (HenryTraill) [reply pages/ahc-categories/list-all-ad-hoc-charge-categories.md:4]: no

File: pages/ahc-categories/ad-hoc-charge-category-object.md [hunk 17]
```diff
@@ -1,4 +1,4 @@
-### Get an Ad Hoc Charge Category
+### Ad Hoc Charge Category Object
 
-Returns the details of an existing ah hoc charge category. You only need to specify the unique
-`id` of the ad hoc charge category to get the correct details.
+Returns the details of an existing Ah Hoc Charge Category. You only need to specify the unique
+`id` of the Ah Hoc Charge Category to get the correct details.
```
Comment (alextford11) [inline pages/ahc-categories/ad-hoc-charge-category-object.md:4]: !

File: pages/adhoccharges/delete-an-ad-hoc-charge.json [hunk 18]
```diff
@@ -1,3 +1,3 @@
 {
-  "message": "Successfully deleted Ad Hoc Charge 37: 37: Tax Law Book"
+  "message": "Successfully deleted Ad Hoc Charge 37: 37: Registration fee"
```
Comment (alextford11) [inline pages/adhoccharges/delete-an-ad-hoc-charge.json:2]: 2 `37:`?

**Review by alextford11 — CHANGES_REQUESTED** [review]

**Pull request merged**